assets/dev/js/editor/components/icons-manager/classes/icon-library.js

```diff
@@ -0,0 +1,152 @@
+import Store from './store';
+
+const IconLibrary = class {
+	loaded = {};
+
+	notifyCallback = null;
+
+	enqueueCSS = ( url ) => {
+		return new Promise( function( resolve ) {
+			if ( ! document.querySelector( 'link[href="' + url + '"]' ) ) {
+				const link = document.createElement( 'link' );
+				link.type = 'text/css';
+				link.rel = 'stylesheet';
+				link.href = url;
+				link.onload = function() {
+					resolve();
+				};
+				const headScript = document.querySelector( 'head' );
+				headScript.append( link );
+			}
+		} );
+	};
+
+	parseCSS = ( css, prefix = 'fa-', displayPrefix = '', library = false ) => {
+		const iconPattern = new RegExp( '\\.' +	prefix + "([^\\.!:]*)::?before\\s*{\\s*content:\\s*[\"|']\\\\[^'|\"]*[\"|'];?\\s*}", 'g' ),
+			index = 1,
+			icons = [];
+		let icon,
+			match = iconPattern.exec( css );
+
+		while ( match ) {
+			icon = {
+				prefix: prefix,
+				selector: prefix + match[ index ].trim( ':' ),
+				name: elementorCommon.helpers.firstLetterUppercase( match[ index ] )
```

Comment:
We have agreed of merging this method with `ucwords`

Reply:
note to self: renaming

Reply:
renamed to upperCaseWords